### ISSUE TITLE
When "replace" is complete ... e.g., the new image is ready

### ISSUE THREAD

**Issue by MonteShaffer** [fengyuanchen/cropper#675]
So I am using "image onload" to get the details of when the image is loaded into the SRC via JQUERY/DOM, however, it appears that loading into cropper memory is something altogether.

```
var $image = $('.img-container > img');
			$image.cropper('replace', "/_images/scans/"+page_id+".jpg");	
```

How do I know the "cropper" is ready to go with the new image loaded?

There needs to be some sort of callback with the replace function, something like ...

```
$image.cropper('replace', "/_images/scans/"+page_id+".jpg",functionCallback);	
```

**Comment by sthomp** [comment]
Check docs for `built`. When you construct your new Cropper() you can implement
`built: function(e){ /* its ready */}`

**Comment by MonteShaffer** [comment]
Yes, indeed the logic is frightful.

```replace``` actually clears the previous data and rebuilds the system so the "build" function may work.

```
this.options.data = null;
          this.load(url);
```

So why don't I just destroy and create myself? 

The build function has

```
// Unbuild first when replace
      if (this.isBuilt) {
        this.unbuild();
      }
```

The logic is a bit circular and hard to follow.  The word "replace" means I change the image, not I sort-of destroy-create with unbuild/build...

Too bad JCrop has died on the vine, the logic of that system is so much more intuitive.

**Comment by MonteShaffer** [comment]
I am also baffled why the image data loads via 

```xhr = new XMLHttpRequest();```

and not something more traditional with a on.load functionality...

```
var img = new Image();
			img.onload = function() {
// I know the image is loaded so I can create/rebuild or whatever ...
}
			img.src = 'sasha-small.jpg';
```

**Comment by fengyuanchen** [comment]
@MonteShaffer Just restart the cropper if you need.

The  `xhr = new XMLHttpRequest();` is for loading image as ArrayBuffer to read Exif Orientation info. You can set the `checkOrientation` option to `false` to skip it.
